core/src/test/java/com/linecorp/armeria/server/ServerChildChannelPipelineCustomizerTest.java

```diff
@@ -0,0 +1,120 @@
+    static ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.http(0);
+            sb.https(0);
+            sb.tlsSelfSigned();
+            sb.childChannelPipelineCustomizer(
+                    pipeline ->  pipeline.addFirst(new ChannelTrafficShapingHandler(1024, 0)));
+            sb.service("/", new AbstractHttpService() {
+                @Override
+                protected HttpResponse doGet(ServiceRequestContext ctx, HttpRequest req)
+                        throws Exception {
+                    return HttpResponse.of(
+                        ResponseHeaders.of(HttpStatus.OK, "header1", Strings.repeat("a", 2048)));
+                }
+            });
+        }
+    };
+
+    static final SSLEngine sslEngine =  SslContextUtil.createSslContext(
+            SslContextBuilder::forClient,
+            /* forceHttp1 */ true,
+            /* tlsAllowUnsafeCiphers */ false,
+            ImmutableList.of()).newEngine(ByteBufAllocator.DEFAULT);
+
+    @RegisterExtension
+    static ServerExtension server2 = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.http(8080);
+            sb.childChannelPipelineCustomizer(
+                    pipeline ->  pipeline.addFirst(new SslHandler(sslEngine)));
+            sb.service("/", (ctx, req) -> HttpResponse.of("content"));
+        }
+    };
+
+    @EnumSource(value = SessionProtocol.class, names = {"H1", "H1C"})
+    @ParameterizedTest
+    void testResponseTimeout(SessionProtocol protocol) {
+        final RequestHeaders requestHeaders = RequestHeaders.of(HttpMethod.GET, "/");
+        final ClientFactory clientFactory = ClientFactory.builder()
+                                                         .tlsNoVerify()
+                                                         .build();
+
+        // using h1 or h1c since http2 compresses headers
+        assertThatThrownBy(() -> WebClient.builder(server.uri(protocol))
+                                          .responseTimeoutMillis(1000)
+                                          .factory(clientFactory)
+                                          .build()
+                                          .blocking()
+                                          .execute(requestHeaders, "content"))
+                .isInstanceOf(ResponseTimeoutException.class);
+    }
+
+    @Test
+    void testSessionProtocolNegotiationException() {
+        final RequestHeaders requestHeaders = RequestHeaders.of(HttpMethod.GET, "/");
+
+        // using h2c
+        assertThatThrownBy(() -> WebClient.builder(server2.uri(SessionProtocol.H2C))
+                 .build()
+                 .blocking()
+                 .execute(requestHeaders, "content"))
+        .isInstanceOf(UnprocessedRequestException.class)
+        .hasCauseInstanceOf(SessionProtocolNegotiationException.class);
+    }
```

Comment:
Again, we want to make sure it's possible to use a customizer to turn a plaintext HTTP server into an HTTPS server by adding an `SslHandler`, so we probably don't need this test case?